### DISCUSSION TITLE
Loss Calculation for Validation

### DISCUSSION THREAD

**Discussion by hilmicanguven** [Project-MONAI/MONAI#2295] [Q&A]
Can anybody tell me where we compute validation (evaluation) loss exactly? I saw that we have evaluator class and make predictions there. This class returns "image", "label", "prediciton", but I didn't see that where we compute  dice loss for evaluation. Thank you for your responses.

**Comment by Nic-Ma** [comment]
Hi @hilmicanguven ,

Thanks for your interest here.
If you are using MONAI evaluator, you can refer to this example:
https://github.com/Project-MONAI/tutorials/blob/master/modules/engines/unet_evaluation_dict.py#L108
If you are writing a pure PyTorch program, please refer to:
https://github.com/Project-MONAI/tutorials/blob/master/3d_segmentation/torch/unet_evaluation_dict.py#L93

Thanks.

**Reply by hilmicanguven** [reply]
Sir @Nic-Ma , thank you first.
I try to implement that tutorial : https://github.com/Project-MONAI/tutorials/tree/master/3d_segmentation/challenge_baseline
Therefore, I use monai library and its evaluator class. I think I use source code of monai library. In order to edit this evaluator, I add monai to my directory and I want to edit this. In my case, I want to change my validation arguments. But I can't find where we exactly calculate dice loss in evaluation. (In training, we use dice loss in run_net.py and I can simply change it, but in evaluation, I can't find we we call the exact function to calculate loss)

**Reply by Nic-Ma** [reply]
May I know what's the `evaluation` program you refer to?

Thanks.

**Reply by hilmicanguven** [reply]
Of course sir,
The code I mentioned, run_net.py, uses two class (trainer and evaluator) and you can find here: https://github.com/Project-MONAI/MONAI/tree/dev/monai/engines
The code use trainer.py to train, and evaluator.py to validation.  Exact class for evaluation is this: https://github.com/Project-MONAI/MONAI/blob/c1e06f5ada1e66a544965136d5029c0bf1f4d510/monai/engines/evaluator.py#L126 
But still I can't find where we calculate evaluation loss.
Thank you

**Reply by Nic-Ma** [reply]
This line set the metrics for validation:
https://github.com/Project-MONAI/tutorials/blob/master/3d_segmentation/challenge_baseline/run_net.py#L188

Thanks.
